plugin/lighthouse/handler_test.go

```diff
@@ -64,29 +65,181 @@ func testDnsServer() {
 		})
 	})
 
-	When("DNS query for typeA record", func() {
+	When("TypeA DNS query for service that doesn't exist", func() {
 		BeforeEach(func() {
-			tc = tests[1]
+			tc = test.Case{
+				// service does not exist, expect error.
+				Qname: "mysvc2.default.svc.cluster.local.",
+				Qtype: dns.TypeA,
+				Rcode: dns.RcodeNameError,
+			}
 		})
-		It("Should return an error for non existent query", func() {
+		It("Should return RcodeNameError", func() {
 			m := tc.Msg()
-			rec := dnstest.NewRecorder(&test.ResponseWriter{})
 			code, err := lh.ServeDNS(ctx, rec, m)
 			Expect(err).To(HaveOccurred())
 			Expect(code).Should(Equal(tc.Rcode))
 		})
 	})
 
-	When("DNS query for typeAAAA record", func() {
+	When("Type A DNS query for pod", func() {
 		BeforeEach(func() {
-			tc = tests[2]
+			tc = test.Case{
+				// service does not exist, expect error.
+				Qname: "mysvc2.default.pod.cluster.local.",
+				Qtype: dns.TypeA,
+				Rcode: dns.RcodeNameError,
+			}
 		})
-		It("Should return an error", func() {
+		It("Should return RcodeNameError", func() {
 			m := tc.Msg()
-			rec := dnstest.NewRecorder(&test.ResponseWriter{})
 			code, err := lh.ServeDNS(ctx, rec, m)
 			Expect(err).To(HaveOccurred())
 			Expect(code).Should(Equal(tc.Rcode))
 		})
 	})
+
+	When("Type AAAA DNS query", func() {
+		BeforeEach(func() {
+			tc = test.Case{
+				// service does not exist, expect error.
+				Qname: "mysvc.default.svc.cluster.local.",
+				Qtype: dns.TypeAAAA,
+				Rcode: dns.RcodeNotImplemented,
+			}
+		})
+		It("Should return error RcodeNotImplemented", func() {
+			m := tc.Msg()
+			code, err := lh.ServeDNS(ctx, rec, m)
+			Expect(err).To(HaveOccurred())
+			Expect(code).Should(Equal(tc.Rcode))
+		})
+	})
+
+	When("Write msg fails", func() {
```

Comment:
_"Write msg fails"_ does not match what it actually tests/verifies below.

Reply:
Oops! Copy paste error, forgot to change msg.